src/UCTSearch.cpp

```diff
@@ -265,18 +274,20 @@ Move UCTSearch::think() {
             increment_playouts();
         }
 
-        Time elapsed;
-        int centiseconds_elapsed = Time::timediff(start, elapsed);
+        // assume nodes = 1.8 ^ depth.
+        int   depth = log(float(m_nodes)) / log(1.8);
```

Comment:
I kind of like the update by time, as that seems a bit more natural for a UCT based search.  GUIs should be fine receiving multiple updates for the same depth.

Reply:
Yes, the GUIs are fine I think (at least xboard was). However, I played a bit around with various timings for output, but couldn't get something that looked nice with a simple linear rule (i.e. output after X ms), it is either too little at the beginning or too much as the end. So, I could make it log(timediff), but at that point I decided that's essentially 'depth'. I don't feel strongly about it, but I have a slight preference for depth based. Up to you to make the choice.

Reply:
Ok, makes sense!  Let's give it a shot and see how it works.